src/sentry/api/serializers/rest_framework/sentry_app.py

```diff
@@ -34,3 +34,20 @@ class SentryAppSerializer(Serializer):
     redirectUrl = serializers.URLField(required=False)
     isAlertable = serializers.BooleanField(required=False)
     overview = serializers.CharField(required=False)
+
+    def validate_events(self, attrs, source):
+        if not attrs.get('scopes'):
+            return attrs
+        resources = [s.split(':')[0] for s in attrs['scopes']]
+        for event in attrs[source]:
+            if event == 'issue':
+                resource = 'event'
+            else:
+                resource = event
+
+            if resource not in resources:
+                raise ValidationError(
+                    u"resource type '{}' does not have the correct permissions.".format(event),
```

Comment:
What if we changed the error here to something more specific like "{event} webhooks require the {scope} permission." (See comment below for where `scope` would come from).